src/editor/editor-dialog.ts

```diff
@@ -0,0 +1,118 @@
+import { LitElement, html, css } from "lit";
+import { createRef, Ref, ref } from "lit/directives/ref.js";
+import { customElement, property } from "lit/decorators.js";
+import "@material/mwc-dialog";
+import "@material/mwc-button";
+import "@material/mwc-list/mwc-list-item.js";
+import "./esphome-editor";
+import { esphomeDialogStyles } from "../styles";
+import { ESPHomeEditor } from "./esphome-editor";
+import { openInstallChooseDialog } from "../install-choose";
+
+@customElement("esphome-editor-dialog")
+class ESPHomeEditorDialog extends LitElement {
+  @property() public fileName!: string;
+
+  editorRef: Ref<ESPHomeEditor> = createRef();
+
+  protected render() {
+    return html`<mwc-dialog
+      open
+      heading=${`Edit ${this.fileName}`}
+      scrimClickAction
+      @closed=${this._handleClose}
+    >
+      <esphome-editor
+        configuration=${this.fileName}
+        ${ref(this.editorRef)}
+        @save=${this._saveFile}
+      ></esphome-editor>
+      <mwc-button
+        slot="secondaryAction"
+        label="Save"
+        @click=${this._saveFile}
+      ></mwc-button>
+      <mwc-button
+        slot="secondaryAction"
+        label="Install"
+        @click=${this.handleInstall}
+      ></mwc-button>
+      <mwc-button
+        slot="secondaryAction"
+        dialogAction="close"
+        label="Close"
+      ></mwc-button>
+    </mwc-dialog> `;
+  }
+
+  static styles = [
+    esphomeDialogStyles,
+    css`
+      :host {
+        --height-header-footer-padding: 152px;
+      }
+      mwc-dialog {
+        --mdc-dialog-min-width: 95vw;
+        --mdc-dialog-max-width: 95vw;
+      }
+
+      esphome-remote-process {
+        height: calc(90vh - var(--height-header-footer-padding));
+      }
+
+      @media only screen and (max-width: 450px) {
+        esphome-remote-process {
+          height: calc(
+            90vh - var(--height-header-footer-padding) -
+              env(safe-area-inset-bottom)
+          );
+          margin-left: -24px;
+          margin-right: -24px;
+        }
+      }
+    `,
+  ];
+
+  private _handleClose() {
+    this.parentNode!.removeChild(this);
+  }
+
+  private handleInstall() {
+    this._saveFile();
+    this._handleClose();
+    openInstallChooseDialog(this.fileName);
+  }
+
+  private _saveFile() {
+    const code = this.editorRef.value!.getValue();
+
+    fetch(`./edit?configuration=${this.fileName}`, {
```

Comment:
All data handling should be moved to https://github.com/esphome/dashboard/tree/main/src/api